Feature/component-organization

rh9891 opened the pull request:
Created new directories for the components based on features (AddTapModal, ClosestTap, Food, FoodFilter, etc.) and reorganized the files. Made additional directories to organize the actions and helper files. Modified how files were being imported. All is working as expected.

rh9891 committed:
Created new folders for the components based on features and reorganized the files.

rh9891 committed:
Finalized the organization of the component structure. Made additional directories for the actions and helper file. Modified how files were being imported. All is working as expected.

nwyin commented on src/App.js:
```diff
@@ -4,35 +4,34 @@ import { Switch, Route } from "react-router";
-import Contribute from "./components/pages/Contribute";
-import Head from "./components/Head";
+import MapPage from "./components/MapPage/MapPage";
+import Mission from "./components/Pages/Mission";
```
I'm getting this error when running locally: 
![image](https://user-images.githubusercontent.com/19523341/145303818-8740d46c-4721-4318-9bfc-eb26638a31af.png)

Seems like the directory is named `pages` but the import uses `Pages` -- resulting in this error. Changing it to `pages` resolves the error

nwyin replied:
Fixed in https://github.com/phlask/phlask-map/pull/222/commits/17867d7015463627225ae851af75c5d0e5970dc1

nwyin commented:
@rh9891 Left a comment with a bug I found. Otherwise, the changes seem to be working as intended. Thanks for the great work!

If you want, we can set up a meeting to talk about PR/code review practices and maybe loop @gcardonag to talk about things we might want to work into the Phlask review workflow.

gcardonag commented:
This MR looks good overall! Thanks for putting in all that time to clean up our file structure. Its already way easier on the eyes locally. 😄 

Regarding the conflicts reported in src/components/Toolbar.js, it seems those are because this branch is still using the older `AddTapModal` component. The following steps will fix your branch locally in case you want to take a jab at solving Git conflicts.
1. Run `git pull`
2. Run `git checkout feature/component-organization`
3. Run `git merge develop`
4. Run `git status` to see the conflict details, there should be 3 files reported.
5. Fix the conflict in `Toolbar.js` by using the version of import that goes `import { AddResourceModal } from "../AddResourceModal";`
6. Run `git add src/components/*` to accept the final version of the conflicted files.
7. Run `git commit` to resolve the conflict and then push up the changes.

If these steps don't work or you have any questions, reach out to me and I can give you a hand or help solve the conflicts on my end once the `Pages` error is resolved.

rh9891 committed:
Made changes to resolve all git conflicts.

rh9891 committed:
Modified import line in order to resolve console error.

rh9891 committed:
Deleted commented code in order to rebuild.

Commit:
rename dir

TIL dir casing matters for git

https://stackoverflow.com/questions/11183788/in-a-git-repository-how-to-properly-rename-a-directory
